src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqBurdenEstimateRepository.kt

```diff
@@ -86,40 +89,84 @@ class JooqBurdenEstimateRepository(
+            val data = ByteArrayOutputStream().use { stream ->
+                PostgresCopyWriter(stream).use { writer ->
+                    for (estimate in estimates)
+                    {
+                        if (estimate.disease != expectedDisease)
+                        {
+                            throw InconsistentDataError("Provided estimate lists disease as '${estimate.disease}' but scenario is for disease '$expectedDisease'")
+                        }
+                        if (estimate.country !in countries)
+                        {
+                            throw UnknownObjectError(estimate.country, "country")
+                        }
+
+                        writer.writeRecord(newBurdenEstimateRecord(setId, estimate, cohortSizeId, estimate.cohortSize))
+                        for (outcome in estimate.outcomes)
+                        {
+                            val outcomeId = outcomeLookup[outcome.key]
+                                    ?: throw UnknownObjectError(outcome.key, "burden-outcome")
+                            writer.writeRecord(newBurdenEstimateRecord(setId, estimate, outcomeId, outcome.value))
+                        }
+                    }
+                }
+                stream.toByteArray()
             }
-            val cohortSize = newBurdenEstimateRecord(setId, estimate, cohortSizeId,
-                   estimate.cohortSize)
 
-            val otherOutcomes = estimate.outcomes.map { outcome ->
-                val outcomeId = outcomeLookup[outcome.key]
-                    ?: throw UnknownObjectError(outcome.key, "burden-outcome")
-                newBurdenEstimateRecord(setId, estimate, outcomeId, outcome.value)
+            // We use dsl.connection to drop down from jOOQ to the JDBC level so we can use CopyManager.
+            dsl.connection { connection ->
+                ByteArrayInputStream(data).use { stream ->
+                    val t = BURDEN_ESTIMATE
+                    val manager = CopyManager(connection as BaseConnection)
+                    manager.copyInto(BURDEN_ESTIMATE, stream, listOf(
+                            t.BURDEN_ESTIMATE_SET,
+                            t.COUNTRY,
+                            t.YEAR,
+                            t.AGE,
+                            t.STOCHASTIC,
+                            t.BURDEN_OUTCOME,
+                            t.VALUE
+                    ))
+                }
             }
-            listOf(cohortSize) + otherOutcomes
         }
-        dsl.batchStore(records).execute()
     }
 
+    private fun getAllCountryIds() = dsl.select(COUNTRY.ID)
+            .from(COUNTRY)
+            .fetch()
+            .map { it[COUNTRY.ID] }
+            .toHashSet()
+
     private fun newBurdenEstimateRecord(
             setId: Int,
             estimate: BurdenEstimate,
             outcomeId: Int,
             outcomeValue: BigDecimal?
-    ): BurdenEstimateRecord
+    ): List<Any?>
```

Comment:
can we name this function in a way that reflects the new output type? like `listOfBurdenEstimateValues` (or something better)?